taskchampion/lib/src/task.rs

```diff
@@ -760,7 +760,9 @@ pub unsafe extern "C" fn tc_task_remove_annotation(task: *mut TCTask, entry: i64
     wrap_mut(
         task,
         |task| {
-            task.remove_annotation(Utc.timestamp(entry, 0))?;
+            if let LocalResult::Single(ts) = Utc.timestamp_opt(entry, 0) {
+                task.remove_annotation(ts)?;
+            }
```

Comment:
AIUI LocalResult::Ambiguous occurs when a timezone has "savings time" and, for example, there are two different times known as "2am".  Because we're explicitly using Utc here, and Utc doesn't do the "savings time" thing, that should never occur.

Reply:
Since that condition should never occur, wouldn't it be better to panic than to return `Ok`?

Reply:
TBH that's what `unreachable!()` is for (which, yes, does panic).  Do you want to make a PR to do that?

Reply:
Sure! I hadn't heard of `unreachable!()` so that will be a good learning opportunity.

Reply:
So my first thought was to wrap this up into a function:

```rs
pub fn utc_timestamp(secs: i64) -> Option<Timestamp> {
    match Utc.timezone_opt(secs, 0) {
        LocalResult::Single(tz) => tz,
        None => None,
        LocalResult::Ambiguous => unreachable!("We're requesting UTC so there shouldn't be any daylight saving time."),
    }
}
```

Then I realized that there is no pattern for sharing utility functions like that between crates. Should there be a sixth "private-utils" crate for stuff like this? I figure there's probably a good reason you haven't done something like that already. Is it better to just copy the `match` block?

Reply:
`taskchampion-lib` already depends on `taskchampion`, so maybe the best plan is to put this utility function in the `taskchampion` crate, maybe in a `time` sub-module.  Then `taskchampion-lib` can call it from there.